pyproject.toml

```diff
@@ -1,10 +1,257 @@
+]
+audio = [
+  "pyworld<=0.2.12",
+  "espnet",
+  "espnet-model-zoo",
+  "pydub",
+]
+beir = [
+  "beir; platform_system != 'Windows'",
+]
+crawler = [
+  "selenium>=4.0.0,!=4.1.4",  # Avoid 4.1.4 due to https://github.com/SeleniumHQ/selenium/issues/10612
+  "webdriver-manager",
+]
+preprocessing = [
+  "beautifulsoup4",
+  "markdown",
+  "python-magic; platform_system != 'Windows'",  # Depends on libmagic: https://pypi.org/project/python-magic/
+  "python-magic-bin; platform_system == 'Windows'",  # Needs to be installed without python-magic, otherwise Windows CI gets stuck.
+]
+ocr = [
+  "pytesseract==0.3.7",
+  "pillow",
+  "pdf2image==1.14.0",
+]
+onnx = [
+  "onnxruntime",
+  "onnxruntime_tools",
+]
+onnx-gpu = [
+  "onnxruntime-gpu",
+  "onnxruntime_tools",
+]
+ray = [
+  "ray>=1.9.1,<2; platform_system != 'Windows'",
+  "ray>=1.9.1,<2,!=1.12.0; platform_system == 'Windows'",  # Avoid 1.12.0 due to https://github.com/ray-project/ray/issues/24169 (fails on windows)
+  "aiorwlock>=1.3.0,<2",
+]
+colab = [
+  "grpcio==1.47.0",
+  "requests>=2.25",  # Needed to avoid dependency conflict with crawler https://github.com/deepset-ai/haystack/pull/2921
+]
+dev = [
+  "pre-commit",
+  # Type check
+  "mypy",
+  "typing_extensions; python_version < '3.8'",
+  # Test
+  "pytest",
+  "pytest-custom_exit_code",  # used in the CI
+  "responses",
+  "tox",
+  "coverage",
+  "python-multipart",
+  "psutil",
+  # Linting
+  "pylint",
+  # Code formatting
+  "black[jupyter]==22.6.0",
+  # Documentation
+  "pydoc-markdown==4.5.1",  # FIXME Unpin!
+  # azure-core is a dependency of azure-ai-formrecognizer
+  # In order to stop malicious pip backtracking during pip install farm-haystack[all] documented in https://github.com/deepset-ai/haystack/issues/2280
+  # we have to resolve a dependency version conflict ourself.
+  # azure-core>=1.23 conflicts with pydoc-markdown's dependency on databind>=1.5.0 which itself requires typing-extensions<4.0.0
+  # azure-core>=1.23 needs typing-extensions>=4.0.1
+  # pip unfortunately backtracks into the databind direction ultimately getting lost.
+  "azure-core<1.23",
+  "mkdocs",
+  "jupytercontrib",
+  "watchdog",  # ==1.0.2
+  "requests-cache",
+]
+test = [
+  "farm-haystack[docstores,audio,crawler,preprocessing,ocr,ray,dev]",
+]
+all = [
+  "farm-haystack[docstores,audio,crawler,preprocessing,ocr,ray,dev,onnx,beir]",
+]
+all-gpu = [
+  "farm-haystack[docstores-gpu,audio,crawler,preprocessing,ocr,ray,dev,onnx-gpu,beir]",
+]
+
+[project.urls]
+"CI: GitHub" = "https://github.com/deepset-ai/haystack/actions"
+"Docs: RTD" = "https://haystack.deepset.ai/overview/intro"
+"GitHub: issues" = "https://github.com/deepset-ai/haystack/issues"
+"GitHub: repo" = "https://github.com/deepset-ai/haystack"
+Homepage = "https://github.com/deepset-ai/haystack"
+
+[tool.hatch.version]
+path = "VERSION.txt"
+pattern = "(?P<version>.+)"
+
+[tool.hatch.build.targets.sdist]
+include = [
+  "/haystack",
+  "/rest_api",
```

Comment:
I don't think we need to ship the REST API with the package (probably now they're shipped even if they shouldn't). On the other hand, we should rather include `LICENSE`?